src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenConverter.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.IdentityModel.JsonWebTokens;
+
+namespace System.IdentityModel.Tokens.Jwt
+{
+    /// <summary>
+    /// Static class to converts an object to a <see cref="JwtSecurityToken"/>
+    /// </summary>
+    public static class JwtSecurityTokenConverter
+    {
+        /// <summary>
+        /// Initializes a new instance of a <see cref="JwtSecurityToken"/> from a <see cref="JsonWebToken"/>
+        /// </summary>
+        /// <param name="token">A JSON Web Token to convert from.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="token"/> is null</exception>
+        public static JwtSecurityToken Convert(JsonWebToken token)
+        {
+            if (token == null)
+                throw new ArgumentNullException(nameof(token));
+
+            if (token.InnerToken != null)
+            {
+                var outerToken = new JwtSecurityToken(token.EncodedToken);
+                outerToken.InnerToken = new JwtSecurityToken(token.InnerToken.EncodedToken);
+                return outerToken;
+            }
+            else if (!string.IsNullOrEmpty(token.EncodedToken))
+            {
+                return new JwtSecurityToken(token.EncodedToken);
+            }
+            else if (token.Header != null && token.Payload != null)
+            {
+                var header = JwtHeader.Deserialize(token.Header.RootElement.GetRawText());
+                var payload = JwtPayload.Deserialize(token.Payload.RootElement.GetRawText());
+                return new JwtSecurityToken(header, payload);
+            }
+
+            throw new ArgumentException("token.EncodedToken or token.Header and token.Payload must be set");
```

Comment:
>token [](http://example.com/codeflow?start=41&length=5)

Should we include an aka.ms link which we can later populate w/details on how a dev should address this error?

Reply:
@jennyf19  if we fix the bug in Wilson https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2065

Reply:
I see. Thanks.